### PR TITLE
Labels: Add 10px padding around bar labels

### LINKED ISSUE #3196: Bar labels should be more conservative about requesting space
Instead of drawing labels when text barely fits within a bar, I'd like labels to request 10px of extra padding on all sides before being drawn. This requires users to zoom in or create less dense charts in order to see labels. Currently, it is too easy to draw plots with too much visual complexity with labels.

Note that this is a small visual breaking change because some charts will now hide labels more often.

### PR DESCRIPTION
**Pull request by CalvinFernandez**
closes #3196

### CODE REVIEW

**Review by hellochar — APPROVED** [review]

File: src/plots/barPlot.ts [hunk 1]
```diff
@@ -584,7 +585,9 @@ namespace Plottable.Plots {
         };
         writer.write(text, containerWidth, containerHeight, writeOptions);
 
-        let tooWide = this._isVertical ? barWidth < measurement.width : barHeight < measurement.height;
+        let tooWide = this._isVertical
+          ? barWidth < (measurement.width + Bar._LABEL_PADDING * 2)
```
Comment (adidahiya) [inline src/plots/barPlot.ts:589]: maybe a comment to help clarify that we want the padding applied on _both_ sides of the text? otherwise lgtm

**Review by adidahiya — APPROVED** [review]

File: src/plots/barPlot.ts (hunk 1 above)
Comment (CalvinFernandez) [reply src/plots/barPlot.ts:589]: word

**Comment by adidahiya** [comment]
@CalvinFernandez does this work for stacked bar sum labels too?

**Comment by CalvinFernandez** [comment]
no this doesn't affect the sum labels

**Comment by adidahiya** [comment]
@CalvinFernandez can you handle that in a follow-up PR? I think the behavior should be consistent across those labels

**Pull request merged**